src/scripts/settings/userSettings.ts

```diff
@@ -106,13 +114,20 @@ export class UserSettings {
      * @param {Object|undefined} config - Configuration or undefined.
      * @return {Object|Promise} Configuration or Promise.
      */
-    serverConfig(config) {
-        const apiClient = this.currentApiClient;
+    serverConfig(): Promise<UserConfiguration>;
+    serverConfig(config: UserConfiguration): Promise<void>;
+    serverConfig(config?: UserConfiguration): Promise<UserConfiguration | void> {
+        // FIXME: correctly handle missing currentApiClient and currentUserId
+        /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+        const apiClient = this.currentApiClient!;
+        /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+        const currentUserId = this.currentUserId!;
```

Comment:
Should we log an error in these cases?

Reply:
I have no preference. I chose to preserve existing behavior for simplicity.